app/charts/shared/chart-dimensions.tsx

```diff
@@ -211,3 +226,60 @@ export const useAxisLabelHeightOffset = ({
     labelWidth,
   };
 };
+
+export const getLongestXLabel = ({
+  xScale,
+  getXLabel,
+  xTimeUnit,
+  formatDate,
+  fontSize,
+}: {
+  xScale: ScaleBand<string>;
+  getXLabel: (d: any) => string;
+  xTimeUnit?: string;
+  formatDate?: (d: any, timeUnit: string) => string;
+  fontSize: number;
+}) => {
+  const domain = xScale.domain();
+
+  const formattedLabels = domain.map((d: any) => {
```

Comment:
@noahonyejese I think `d` is known to be a `string`, as you typed `ScaleBand<string>`? 🤔